src/emulator/modules/SceLibKernel/SceLibKernel.cpp

```diff
@@ -32,8 +34,164 @@
 #include <psp2/io/dirent.h>
 #include <psp2/io/stat.h>
 #include <psp2/kernel/modulemgr.h>
-#include <stdlib.h>
-#include <util/log.h>
+
+#include <algorithm>
+#include <cstdlib>
+
+static const bool LOG_SYNC_OBJECTS = false;
+
+SceUID create_mutex(SceUID* uid_out, HostState& host, SceUID thread_id, MutexPtrs& host_mutexes, const char *name, SceUInt attr, int init_count, bool is_lw) {
+    if ((strlen(name) > 31) && ((attr & 0x80) == 0x80)) {
+        return error("sceKernelCreate(Lw)Mutex", SCE_KERNEL_ERROR_UID_NAME_TOO_LONG);
+    }
+    if (init_count < 0) {
+        return error("sceKernelCreate(Lw)Mutex", SCE_KERNEL_ERROR_ILLEGAL_COUNT);
+    }
+    if (init_count > 1 && (attr & SCE_KERNEL_MUTEX_ATTR_RECURSIVE)) {
+        return error("sceKernelCreate(Lw)Mutex", SCE_KERNEL_ERROR_ILLEGAL_COUNT);
+    }
+    
+    const MutexPtr mutex = std::make_shared<Mutex>();
+    mutex->lock_count = init_count;
+    std::copy(name, name + emu::KERNELOBJECT_MAX_NAME_LENGTH, mutex->name);
+    mutex->attr = attr;
+    mutex->owner = nullptr;
+    if (init_count > 0) {
+        const ThreadStatePtr thread = lock_and_find(thread_id, host.kernel.threads, host.kernel.mutex);
+        mutex->owner = thread;
+    }
+    const std::unique_lock<std::mutex> lock(host.kernel.mutex);
+    const SceUID uid = host.kernel.next_uid++;
+    host_mutexes.emplace(uid, mutex);
+
+    if (LOG_SYNC_OBJECTS) {
+        LOG_DEBUG("Creating {}: uid:{} thread_id:{} name:\"{}\" attr:{} init_count:{}",
+            is_lw ? "lwmutex" : "mutex", uid, thread_id, name, attr, init_count);
+    }
+
+    if (uid_out) {
+        *uid_out = uid;
+    }
+
+    return SCE_KERNEL_OK;
+}
+
+int lock_mutex(HostState& host, SceUID thread_id, MutexPtrs& host_mutexes, SceUID mutexid, int lock_count, unsigned int *timeout, bool is_lw) {
+    assert(lock_count == 1);
+    assert(!timeout);
+
+    // TODO Don't lock twice.
+    const MutexPtr mutex = lock_and_find(mutexid, host_mutexes, host.kernel.mutex);
+    if (!mutex) {
+        return error("sceKernelLockMutex", SCE_KERNEL_ERROR_UNKNOWN_MUTEX_ID);
+    }
+
+    if (LOG_SYNC_OBJECTS) {
+        LOG_DEBUG("Locking mutex: uid:{} thread_id:{} name:\"{}\" attr:{} lock_count:{}",
+            mutexid, thread_id, mutex->name, mutex->attr, mutex->lock_count);
+    }
+
+    const std::unique_lock<std::mutex> lock(mutex->mutex);
+
+    const ThreadStatePtr thread = lock_and_find(thread_id, host.kernel.threads, host.kernel.mutex);
+
+    bool is_recursive = (mutex->attr & SCE_KERNEL_MUTEX_ATTR_RECURSIVE);
+
+    // if (is_lw) {
+
+    if (mutex->lock_count > 0) {
+        // Already owned
+
+        if (mutex->owner == thread) {
+            // Owned by ourselves
+
+            if (is_recursive) {
+                // TODO: Recursive mutex support
```

Comment:
Just increment lock_count

Reply:
Oh, wasn't sure if it's that simple, cool.